--- fceu2.1.5/src/file.cpp~	2010-09-19 02:05:09.000000000 +0200
+++ fceu2.1.5/src/file.cpp	2012-02-12 10:11:53.832509623 +0100
@@ -310,7 +310,7 @@
 				if(magic==0x088b1f) {
 					 // maybe gzip... 
 
-					void* gzfile = gzopen(fileToOpen.c_str(),"rb");
+					gzFile gzfile = gzopen(fileToOpen.c_str(),"rb");
 					if(gzfile) {
 						delete fp;
 
--- fceu2.1.5/SConstruct~	2011-06-04 07:26:55.000000000 +0200
+++ fceu2.1.5/SConstruct	2012-02-12 10:15:10.593425125 +0100
@@ -39,6 +39,8 @@
   env.Append(CCFLAGS = os.environ['CFLAGS'].split())
 if os.environ.has_key('LDFLAGS'):
   env.Append(LINKFLAGS = os.environ['LDFLAGS'].split())
+if os.environ.has_key('LIBS'):
+  env.Append(LIBS = os.environ['LIBS'].split())
 
 print "platform: ", env['PLATFORM']
 
